source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,379 @@ Documents
+
+   BsonDocument author = new BsonDocument()
+       .append("_id", new BsonObjectId())
+       .append("name", new BsonString("Gabriel García Márquez"))
+       .append("lastUpdated", new BsonDateTime(new Date(120, 2, 26).getTime()))
+       .append("novels", new BsonArray(Arrays.asList(
+           new BsonDocument().append("title", new BsonString("One Hundred Years of Solitude")).append("yearPublished", new BsonInt32(1967)),
+           new BsonDocument().append("title", new BsonString("Chronicle of a Death Foretold")).append("yearPublished", new BsonInt32(1981)),
+           new BsonDocument().append("title", new BsonString("Love in the Time of Cholera")).append("yearPublished", new BsonInt32(1985))
+      )));
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``BsonDocument`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<BsonDocument> collection = database.getCollection("authors", BsonDocument.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+
+   import com.mongodb.client.model.Filters;
+
+   // <MongoCollection setup code here>
+
+   BsonDocument doc = collection.find(Filters.eq("name", "Gabriel García Márquez")).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id").getValue()
+               + ", name: " + doc.getString("name").getValue()
+               + ", lastUpdated: " + new Date(doc.getDateTime("lastUpdated").getValue()));
+
+       doc.getArray("novels").forEach((novel) -> {
+           System.out.println("title: " + novel.asDocument().getString("title").getValue()
+                   + ", yearPublished: " + novel.asDocument().getInt32("yearPublished").getValue());
+       });
+   }
+
+.. note::
+
+   The code sample above uses helper methods that check the returned type
+   and throw a :java-docs:`BsonInvalidOperationException </apidocs/bson/org/bson/BsonInvalidOperationException.html>`
+   if it is unable to cast the field value. You can call the
+   :java-docs:`get() </apidocs/bson/org/bson/BsonDocument.html#get(java.lang.Object)>`
+   method specified by the ``Map`` interface to retrieve field values as
+   type :java-docs:`BsonValue </apidocs/bson/org/bson/BsonValue.html>` and to
+   skip type checking.
+
+The output should resemble the following:
+
+.. code-block:: none
+   :copyable: false
+
+   _id: 5fb5fad05f734e3794741a35, name: Gabriel García Márquez, lastUpdated: Thu Mar 26 00:00:00 EDT 2020
```

Comment:
I'm surprised you got 2020 given the date you used above

Reply:
Will update for new field.